From 98ee6e2ac5447154a6d0acaf2fca5173130e23c5 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Tue, 21 Jun 2016 07:10:46 +0200
Subject: [PATCH 02/12] ARM: bcm2835: Add the emmc pin group to bcm283x.dtsi.

Patch-mainline: v4.10-rc1
Git-commit: 21ff843931b2e5a9b628ac56fd0f2e4355890096 (partial)
References: bsc#983145

Acked-by: Michal Marek <mmarek@suse.com>

---
 arch/arm/boot/dts/bcm283x.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/bcm283x.dtsi b/arch/arm/boot/dts/bcm283x.dtsi
index 013fe042325f..9af33d799d07 100644
--- a/arch/arm/boot/dts/bcm283x.dtsi
+++ b/arch/arm/boot/dts/bcm283x.dtsi
@@ -253,6 +253,10 @@
 				brcm,pins = <32 33>;
 				brcm,function = <BCM2835_FSEL_ALT3>;
 			};
+			emmc_gpio48: emmc_gpio48 {
+				brcm,pins = <48 49 50 51 52 53>;
+				brcm,function = <BCM2835_FSEL_ALT3>;
+			};
 			spi1_gpio16: spi1_gpio16 {
 				brcm,pins = <16 17 18 19 20 21>;
 				brcm,function = <BCM2835_FSEL_ALT4>;
-- 
2.6.6

