From 73c019d65eff5c5269aeea519a0be4d0968b2855 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@xxxxxxxxxx>
Date: Wed, 22 Jun 2016 13:42:18 +0200
Subject: [PATCH 5/5] mmc: bcm2835-sdhost: add to devicetree

Patch-mainline: v4.12-rc1
Git-commit: 7f31a955a0c34de0463f7ff50b2bd62a5cce4204 (partial)
References: bsc#983145


Acked-by: Michal Marek <mmarek@suse.com>

---
 arch/arm/boot/dts/bcm283x.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/boot/dts/bcm283x.dtsi b/arch/arm/boot/dts/bcm283x.dtsi
index aeb4cd6850b9..8d1d9888f839 100644
--- a/arch/arm/boot/dts/bcm283x.dtsi
+++ b/arch/arm/boot/dts/bcm283x.dtsi
@@ -219,6 +219,14 @@
 			status = "disabled";
 		};
 
+		sdhost: sdhost@7e202000 {
+			compatible = "brcm,bcm2835-sdhost";
+			reg = <0x7e202000 0x100>;
+			interrupts = <2 24>;
+			clocks = <&clocks BCM2835_CLOCK_VPU>;
+			status = "disabled";
+		};
+
 		hvs@7e400000 {
 			compatible = "brcm,bcm2835-hvs";
 			reg = <0x7e400000 0x6000>;
-- 
2.6.2

