From: Ivan Vecera <cera@cera.cz>
Date: Tue, 10 Jul 2018 22:59:46 +0200
Subject: be2net: remove unused tx_jiffies field from be_tx_stats
Patch-mainline: v4.19-rc1
Git-commit: f9520b86dc22b6ac0ad2926cfa334e9fecb68a12
References: bsc#1104714 FATE#325913

Signed-off-by: Ivan Vecera <cera@cera.cz>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/emulex/benet/be.h |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/net/ethernet/emulex/benet/be.h
+++ b/drivers/net/ethernet/emulex/benet/be.h
@@ -217,7 +217,6 @@ struct be_tx_stats {
 	u64 tx_vxlan_offload_pkts;
 	u64 tx_reqs;
 	u64 tx_compl;
-	ulong tx_jiffies;
 	u32 tx_stops;
 	u32 tx_drv_drops;	/* pkts dropped by driver */
 	/* the error counters are described in be_ethtool.c */
