From: Roman Mashak <mrv@mojatatu.com>
Date: Fri, 2 Mar 2018 20:52:01 -0500
Subject: net sched actions: corrected extack message
Patch-mainline: v4.17-rc1
Git-commit: d143b9e3055358e160d8d5975d889ce5c149625a
References: bsc#1109837

Signed-off-by: Roman Mashak <mrv@mojatatu.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 net/sched/act_api.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/sched/act_api.c
+++ b/net/sched/act_api.c
@@ -1087,7 +1087,7 @@ tcf_add_notify(struct net *net, struct n
 
 	if (tca_get_fill(skb, actions, portid, n->nlmsg_seq, n->nlmsg_flags,
 			 RTM_NEWACTION, 0, 0) <= 0) {
-		NL_SET_ERR_MSG(extack, "Failed to fill netlink attributes while deleting TC action");
+		NL_SET_ERR_MSG(extack, "Failed to fill netlink attributes while adding TC action");
 		kfree_skb(skb);
 		return -EINVAL;
 	}
