From: Ganesh Goudar <ganeshgr@chelsio.com>
Date: Thu, 2 Aug 2018 15:34:52 +0530
Subject: net: Fix coding style in skb_push()
Patch-mainline: v4.19-rc1
Git-commit: 9aba2f801eea5070f1d5588cd4052588437b9eea
References: bsc#1109837

Signed-off-by: Ganesh Goudar <ganeshgr@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 net/core/skbuff.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -1452,7 +1452,7 @@ void *skb_push(struct sk_buff *skb, unsi
 {
 	skb->data -= len;
 	skb->len  += len;
-	if (unlikely(skb->data<skb->head))
+	if (unlikely(skb->data < skb->head))
 		skb_under_panic(skb, len, __builtin_return_address(0));
 	return skb->data;
 }
