From: Tonghao Zhang <xiangxia.m.yue@gmail.com>
Date: Sun, 7 Oct 2018 18:41:50 -0700
Subject: net: vhost: remove bad code line
Patch-mainline: v4.20-rc1
Git-commit: abf1a08ff3237a27188ff8cc2904f2cea893af55
References: bsc#1109837

Signed-off-by: Tonghao Zhang <xiangxia.m.yue@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/vhost/net.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -560,7 +560,6 @@ static int vhost_net_tx_get_vq_desc(stru
 	if (r == tvq->num && tvq->busyloop_timeout) {
 		/* Flush batched packets first */
 		if (!vhost_sock_zcopy(tvq->private_data))
-			// vhost_net_signal_used(tnvq);
 			vhost_tx_batch(net, tnvq, tvq->private_data, msghdr);
 
 		vhost_net_busy_poll(net, rvq, tvq, busyloop_intr, false);
