From: "Milan P. Gandhi" <mgandhi@redhat.com>
Date: Thu, 9 Aug 2018 21:49:24 +0530
Subject: [PATCH] nvme: fix typo in nvme_identify_ns_descs
Git-commit: 53b3a66163ea6cc7a86e0a3a04b1166d96665824
Patch-mainline: v4.20-rc1
References: bsc#1104967,FATE#325924

Signed-off-by: Milan P. Gandhi <mgandhi@redhat.com>
Reviewed-by: Chaitanya Kulkarni <chaitanya.kulkarni@wdc.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 drivers/nvme/host/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/nvme/host/core.c b/drivers/nvme/host/core.c
index e0a9e1c5b30e..f0778d3dd2f8 100644
--- a/drivers/nvme/host/core.c
+++ b/drivers/nvme/host/core.c
@@ -971,7 +971,7 @@ static int nvme_identify_ns_descs(struct nvme_ctrl *ctrl, unsigned nsid,
 			uuid_copy(&ids->uuid, data + pos + sizeof(*cur));
 			break;
 		default:
-			/* Skip unnkown types */
+			/* Skip unknown types */
 			len = cur->nidl;
 			break;
 		}
-- 
2.16.4

