From: Jason Wang <jasowang@redhat.com>
Date: Wed, 17 May 2017 12:14:38 +0800
Subject: skb_array: introduce skb_array_unconsume
Patch-mainline: v4.13-rc1
Git-commit: 3acb696015a222f4b25c1b5dce4e36b2d4980da6
References: bsc#1109837

Signed-off-by: Jason Wang <jasowang@redhat.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 include/linux/skb_array.h |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/include/linux/skb_array.h
+++ b/include/linux/skb_array.h
@@ -156,6 +156,12 @@ static void __skb_array_destroy_skb(void
 	kfree_skb(ptr);
 }
 
+static inline void skb_array_unconsume(struct skb_array *a,
+				       struct sk_buff **skbs, int n)
+{
+	ptr_ring_unconsume(&a->ring, (void **)skbs, n, __skb_array_destroy_skb);
+}
+
 static inline int skb_array_resize(struct skb_array *a, int size, gfp_t gfp)
 {
 	return ptr_ring_resize(&a->ring, size, gfp, __skb_array_destroy_skb);
