From: Egbert Eich <eich@suse.de>
Date: Tue Apr 26 13:52:15 2016 +0200
Subject: vgaarb: Add more context to error messages
Patch-mainline: never, Catch issues in proprietary driver
References: bsc#976868

Signed-off-by: Egbert Eich <eich@suse.de>
Signed-off-by: Egbert Eich <eich@suse.com>
---
 drivers/gpu/vga/vgaarb.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)
diff --git a/drivers/gpu/vga/vgaarb.c b/drivers/gpu/vga/vgaarb.c
index 9abcaa5..8ebd302 100644
--- a/drivers/gpu/vga/vgaarb.c
+++ b/drivers/gpu/vga/vgaarb.c
@@ -636,6 +636,7 @@ static bool vga_arbiter_del_pci_device(struct pci_dev *pdev)
 	 * it's copies of the pci_dev pointer
 	 */
 	vga_arb_device_card_gone(pdev);
+	pr_info("device added: PCI:%s\n", pci_name(pdev));
 
 	/* Wake up all possible waiters */
 	wake_up_all(&vga_wait_queue);
@@ -1100,7 +1101,7 @@ static ssize_t vga_arb_write(struct file *file, const char __user *buf,
 		pr_debug("vgadev %p\n", vgadev);
 		if (vgadev == NULL) {
 			if (pdev) {
-				pr_err("this pci device is not a vga device\n");
+				pr_err("this pci device is not a vga device: %s\n", pci_name(pdev));
 				pci_dev_put(pdev);
 			}
 
