From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 30 Aug 2017 10:32:30 +0200
Subject: [PATCH] Linux 4.12.10
References: bnc#1060662
Patch-mainline: 4.12.10
Git-commit: 6371f030c4dc8b69140819e92803aae7e6039cd6

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index a6c2a5e7a48d..6889ec6a091d 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 12
-SUBLEVEL = 9
+SUBLEVEL = 10
 EXTRAVERSION =
 NAME = Fearless Coyote
 
-- 
2.14.2

