From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 16 Aug 2017 13:47:18 -0700
Subject: [PATCH] Linux 4.12.8
References: bnc#1060662
Patch-mainline: 4.12.8
Git-commit: a0fb6543b40f216f72ef57ff3f61df918047c0fc

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index ebe69a704bca..6da481d08441 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 12
-SUBLEVEL = 7
+SUBLEVEL = 8
 EXTRAVERSION =
 NAME = Fearless Coyote
 
-- 
2.14.2

