From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 13 Sep 2017 14:18:03 -0700
Subject: [PATCH] Linux 4.12.13
References: bnc#1060662
Patch-mainline: 4.12.13
Git-commit: 5d7d2e03e0f01a992e3521b180c3d3e67905f269

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index e96306381ee8..983224467a4d 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 12
-SUBLEVEL = 12
+SUBLEVEL = 13
 EXTRAVERSION =
 NAME = Fearless Coyote
 
-- 
2.14.2

