From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Fri, 11 Aug 2017 08:34:14 -0700
Subject: [PATCH] Linux 4.12.6
References: bnc#1060662
Patch-mainline: 4.12.6
Git-commit: 311d0c83145730eede422215b346d257ca6c0869

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 382e967b0792..c8d80b50495a 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 12
-SUBLEVEL = 5
+SUBLEVEL = 6
 EXTRAVERSION =
 NAME = Fearless Coyote
 
-- 
2.14.2

