From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sun, 10 Sep 2017 07:47:49 +0200
Subject: [PATCH] Linux 4.12.12
References: bnc#1060662
Patch-mainline: 4.12.12
Git-commit: 6ff98e8e5d2445b4d3c13ce2e4f071815041f2d1

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index e7b2b54b032c..e96306381ee8 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 12
-SUBLEVEL = 11
+SUBLEVEL = 12
 EXTRAVERSION =
 NAME = Fearless Coyote
 
-- 
2.14.2

