From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sun, 6 Aug 2017 09:22:44 -0700
Subject: [PATCH] Linux 4.12.5
References: bnc#1060662
Patch-mainline: 4.12.5
Git-commit: 45100eec5f682b606e35ab0567417bf611821a60

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index bfdc92c2e47a..382e967b0792 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 12
-SUBLEVEL = 4
+SUBLEVEL = 5
 EXTRAVERSION =
 NAME = Fearless Coyote
 
-- 
2.14.2

