From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Thu, 27 Jul 2017 15:10:39 -0700
Subject: [PATCH] Linux 4.12.4
References: bnc#1060662
Patch-mainline: 4.12.4
Git-commit: 13df91dbc06ed4323cf8bbc956ec9a0098f1361e

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 45b8746b09f7..bfdc92c2e47a 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 12
-SUBLEVEL = 3
+SUBLEVEL = 4
 EXTRAVERSION =
 NAME = Fearless Coyote
 
-- 
2.14.2

