From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 12 Jul 2017 16:55:36 +0200
Subject: [PATCH] Linux 4.12.1
References: bnc#1060662
Patch-mainline: 4.12.1
Git-commit: cb6621858813522e62fcba835541e4fcf57b3cb3

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 283c6236438e..1286f8cc7b5b 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 12
-SUBLEVEL = 0
+SUBLEVEL = 1
 EXTRAVERSION =
 NAME = Fearless Coyote
 
-- 
2.14.2

