From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Thu, 7 Sep 2017 08:37:34 +0200
Subject: [PATCH] Linux 4.12.11
References: bnc#1060662
Patch-mainline: 4.12.11
Git-commit: 13817987369afc8a65c6d7590cfca57f9da4a9a2

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6889ec6a091d..e7b2b54b032c 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 12
-SUBLEVEL = 10
+SUBLEVEL = 11
 EXTRAVERSION =
 NAME = Fearless Coyote
 
-- 
2.14.2

