From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sat, 15 Jul 2017 13:09:20 +0200
Subject: [PATCH] Linux 4.12.2
References: bnc#1060662
Patch-mainline: 4.12.2
Git-commit: ab35d16f66d63b625396edfba013dd258e13f560

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 1286f8cc7b5b..7c81bbba2943 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 12
-SUBLEVEL = 1
+SUBLEVEL = 2
 EXTRAVERSION =
 NAME = Fearless Coyote
 
-- 
2.14.2

