From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Thu, 24 Aug 2017 17:15:18 -0700
Subject: [PATCH] Linux 4.12.9
References: bnc#1060662
Patch-mainline: 4.12.9
Git-commit: e0e7ae96b72318d3d8797de0befbe6c6f001f663

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6da481d08441..a6c2a5e7a48d 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 12
-SUBLEVEL = 8
+SUBLEVEL = 9
 EXTRAVERSION =
 NAME = Fearless Coyote
 
-- 
2.14.2

