From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sat, 12 Aug 2017 19:34:48 -0700
Subject: [PATCH] Linux 4.12.7
References: bnc#1060662
Patch-mainline: 4.12.7
Git-commit: 1ea1c41db9839e56eb0f41dfffc7c6dd80bf7a64

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index c8d80b50495a..ebe69a704bca 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 12
-SUBLEVEL = 6
+SUBLEVEL = 7
 EXTRAVERSION =
 NAME = Fearless Coyote
 
-- 
2.14.2

