From: Christoph Hellwig <hch@lst.de>
Date: Tue, 29 Aug 2017 18:48:38 +0200
Subject: [PATCH] bsg: remove #if 0'ed code
Git-commit: c529594f93ae64de2a84e7fff903ae6844664912
Patch-mainline: v4.14-rc1
References: bsc#1104967,FATE#325924

Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 block/bsg.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/block/bsg.c b/block/bsg.c
index 37663b664666..ee1335c68de7 100644
--- a/block/bsg.c
+++ b/block/bsg.c
@@ -932,15 +932,8 @@ static long bsg_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 
 		return ret;
 	}
-	/*
-	 * block device ioctls
-	 */
 	default:
-#if 0
-		return ioctl_by_bdev(bd->bdev, cmd, arg);
-#else
 		return -ENOTTY;
-#endif
 	}
 }
 
-- 
2.16.4

