From: Jeff Layton <jlayton@kernel.org>
Date: Tue, 23 Apr 2019 14:18:45 -0400
Subject: ceph: fix NULL pointer deref when debugging is enabled
Git-commit: 40e7e2c0e86464bca839cdf891bd58a6d41b60b4
Patch-mainline: v5.2-rc1
References: bsc#1138681

Signed-off-by: Jeff Layton <jlayton@kernel.org>
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 fs/ceph/file.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/ceph/file.c
+++ b/fs/ceph/file.c
@@ -926,7 +926,7 @@ ceph_direct_read_write(struct kiocb *ioc
 
 	dout("sync_direct_%s on file %p %lld~%u snapc %p seq %lld\n",
 	     (write ? "write" : "read"), file, pos, (unsigned)count,
-	     snapc, snapc->seq);
+	     snapc, snapc ? snapc->seq : 0);
 
 	ret = filemap_write_and_wait_range(inode->i_mapping, pos, pos + count);
 	if (ret < 0)
