From: Steve French <stfrench@microsoft.com>
Date: Sun, 31 Mar 2019 18:00:54 -0500
Subject: [PATCH] cifs: update module internal version number
Git-commit: cb4f7bf6be10b35510e6b2e60f80d85ebc22a578
Patch-mainline: v5.2-rc1
References: bsc#1144333

To 2.20

Signed-off-by: Steve French <stfrench@microsoft.com>
Acked-by: Paulo Alcantara <palcantara@suse.de>
---
 fs/cifs/cifsfs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/cifs/cifsfs.h b/fs/cifs/cifsfs.h
index c47d93d74d75..aea005703785 100644
--- a/fs/cifs/cifsfs.h
+++ b/fs/cifs/cifsfs.h
@@ -152,5 +152,5 @@ extern long cifs_ioctl(struct file *filep, unsigned int cmd, unsigned long arg);
 extern const struct export_operations cifs_export_ops;
 #endif /* CONFIG_CIFS_NFSD_EXPORT */
 
-#define CIFS_VERSION   "2.19"
+#define CIFS_VERSION   "2.20"
 #endif				/* _CIFSFS_H */
-- 
2.22.0

