From ee66bf4c99cc8cf3dc12e8cf4b8c348bf9ff4905 Mon Sep 17 00:00:00 2001
From: Michal Suchanek <msuchanek@suse.de>
Date: Tue, 10 Apr 2018 09:55:38 +0200
Subject: [PATCH] KABI: hide ftrace_enabled in paca

Patch-mainline: no, kabi
References: bsc#1088804

Signed-off-by: Michal Suchanek <msuchanek@suse.de>
---
 arch/powerpc/include/asm/paca.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/powerpc/include/asm/paca.h b/arch/powerpc/include/asm/paca.h
index 3c1846a75a6f..9d738ee1d8f3 100644
--- a/arch/powerpc/include/asm/paca.h
+++ b/arch/powerpc/include/asm/paca.h
@@ -215,7 +215,9 @@ struct paca_struct {
 	u8 hmi_event_available;		/* HMI event is available */
 	u8 hmi_p9_special_emu;		/* HMI P9 special emulation */
 #endif
+#ifndef __GENKSYMS__ /* 2byte hole -> 1byte hole */
 	u8 ftrace_enabled;		/* Hard disable ftrace */
+#endif
 
 	/* Stuff for accurate time accounting */
 	struct cpu_accounting_data accounting;
-- 
2.13.6

