From 3661facd402932aa5ff2fc375cc7389f3175ba82 Mon Sep 17 00:00:00 2001
From: "Lee, Chun-Yi" <jlee@suse.com>
Date: Tue, 18 Aug 2015 12:11:07 +0800
Subject: [PATCH] PM / hibernate: Force verify hibernation signature in signed
 modules environment

Patch-mainline: Never, unless BSD-style securelevel accepted by upstream
References: fate#316350

Signed-off-by: Lee, Chun-Yi <jlee@suse.com>
---
 kernel/power/hibernate.c |   14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

--- a/kernel/power/hibernate.c
+++ b/kernel/power/hibernate.c
@@ -72,7 +72,19 @@ static const struct platform_hibernation
 
 bool hibernation_available(void)
 {
-	return ((nohibernate == 0) && (get_securelevel() <= 0));
+	if (nohibernate != 0)
+		return false;
+
+	if (get_securelevel() <= 0)
+		return true;
+	else if (get_securelevel() > 0) {
+#ifdef CONFIG_HIBERNATE_VERIFICATION
+		sigenforce = 1;
+		return true;
+#else
+		return false;
+#endif
+	}
 }
 
 /**
