From a0eb859d0ac42beb4120b306119d1f6955b528aa Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.com>
Date: Thu, 15 Mar 2018 20:49:17 +0100
Subject: [PATCH] Thunderbolt: kABI paddings added
Patch-mainline: Never (SUSE kABI padding)
References: bsc#1070681

just in case

Signed-off-by: Oliver Neukum <oneukum@suse.com>
---
 include/linux/thunderbolt.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/linux/thunderbolt.h b/include/linux/thunderbolt.h
index 5d7ccf0..113a2f1 100644
--- a/include/linux/thunderbolt.h
+++ b/include/linux/thunderbolt.h
@@ -75,6 +75,7 @@ struct tb {
 	const struct tb_cm_ops *cm_ops;
 	int index;
 	enum tb_security_level security_level;
+	void *suse_kabi_padding;
 	unsigned long privdata[0];
 };
 
@@ -97,6 +98,7 @@ static inline unsigned int tb_phy_port_from_link(unsigned int link)
 struct tb_property_dir {
 	const uuid_t *uuid;
 	struct list_head properties;
+	void *suse_kabi_padding;
 };
 
 enum tb_property_type {
@@ -130,6 +132,7 @@ struct tb_property {
 		char *text;
 		u32 immediate;
 	} value;
+	void *suse_kabi_padding;
 };
 
 struct tb_property_dir *tb_property_parse_dir(const u32 *block,
@@ -182,6 +185,7 @@ struct tb_nhi {
 	bool going_away;
 	struct work_struct interrupt_work;
 	u32 hop_count;
+	void *suse_kabi_padding;
 };
 
 /**
@@ -229,6 +233,7 @@ struct tb_ring {
 	u16 eof_mask;
 	void (*start_poll)(void *data);
 	void *poll_data;
+	void *suse_kabi_padding;
 };
 
 /* Leave ring interrupt enabled on suspend */
-- 
2.13.6

