From: Denis Kirjanov <denis.kirjanov@suse.com>
Subject: fix i915 compilation 
References: CVE-2019-0154,bsc#1135966,CVE-2019-0155,bsc#1135967 
Patch-mainline: Never, compilation fix

Signed-off-by: Denis Kirjanov <denis.kirjanov@suse.com>

---

--- a/drivers/gpu/drm/i915/i915_gem_execbuffer.c
+++ b/drivers/gpu/drm/i915/i915_gem_execbuffer.c
@@ -1910,7 +1910,7 @@ shadow_batch_pin(struct i915_execbuffer
 	 */
 	if (CMDPARSER_USES_GGTT(dev_priv)) {
 		flags = PIN_GLOBAL;
-		vm = &dev_priv->ggtt.base;
+		vm = &dev_priv->ggtt.vm;
 	} else if (eb->vm->has_read_only) {
 		flags = PIN_USER;
 		vm = eb->vm;
