From afac6bd6d1a3ea38c800aa82d6d0ceda518b3aec Mon Sep 17 00:00:00 2001
From: Heinz Mauelshagen <heinzm@redhat.com>
Date: Thu, 1 Feb 2018 19:06:12 +0100
Subject: [PATCH] dm unstripe: remove unnecessary header includes
Git-commit: afac6bd6d1a3ea38c800aa82d6d0ceda518b3aec
References: bsc#1111974
Patch-mainline: v4.17-rc1

Signed-off-by: Heinz Mauelshagen <heinzm@redhat.com>
Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Signed-off-by: Coly Li <colyli@suse.de>

---
 drivers/md/dm-unstripe.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/md/dm-unstripe.c b/drivers/md/dm-unstripe.c
index cf7ac073d840..954b7ab4e684 100644
--- a/drivers/md/dm-unstripe.c
+++ b/drivers/md/dm-unstripe.c
@@ -7,12 +7,6 @@
 #include "dm.h"
 
 #include <linux/module.h>
-#include <linux/init.h>
-#include <linux/blkdev.h>
-#include <linux/bio.h>
-#include <linux/slab.h>
-#include <linux/bitops.h>
-#include <linux/device-mapper.h>
 
 struct unstripe_c {
 	struct dm_dev *dev;
-- 
2.19.0

