From: Steve French <stfrench@microsoft.com>
Date: Tue, 21 Aug 2018 21:37:53 -0500
Subject: [PATCH] cifs: update internal module version number for cifs.ko to
 2.12
Git-commit: 7753e38286d2bd7380df8be843de83198e2ce492
Patch-mainline: v4.19-rc1
References: bsc#1144333

Signed-off-by: Steve French <stfrench@microsoft.com>
Acked-by: Aurelien Aptel <aaptel@suse.com>
---
 fs/cifs/cifsfs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/cifs/cifsfs.h b/fs/cifs/cifsfs.h
index f3a78efc3109..f047e87871a1 100644
--- a/fs/cifs/cifsfs.h
+++ b/fs/cifs/cifsfs.h
@@ -148,5 +148,5 @@ extern long cifs_ioctl(struct file *filep, unsigned int cmd, unsigned long arg);
 extern const struct export_operations cifs_export_ops;
 #endif /* CONFIG_CIFS_NFSD_EXPORT */
 
-#define CIFS_VERSION   "2.12"
+#define CIFS_VERSION   "2.13"
 #endif				/* _CIFSFS_H */
-- 
2.16.4

