From 2d77dafe23b6c0cc9e501bda0e3f138b96ecd811 Mon Sep 17 00:00:00 2001
From: Wang Sheng-Hui <shhuiw@foxmail.com>
Date: Mon, 5 Feb 2018 10:25:44 +0800
Subject: [PATCH] dm: remove unused macro DM_MOD_NAME_SIZE
Git-commit: 2d77dafe23b6c0cc9e501bda0e3f138b96ecd811
Patch-mainline: v4.17-rc1
References: bsc#1093023

Signed-off-by: Wang Sheng-Hui <shhuiw@foxmail.com>
Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Signed-off-by: Coly Li <colyli@suse.de>

---
 drivers/md/dm-target.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/md/dm-target.c b/drivers/md/dm-target.c
index c0d7e60820c4..314d17ca6466 100644
--- a/drivers/md/dm-target.c
+++ b/drivers/md/dm-target.c
@@ -16,8 +16,6 @@
 static LIST_HEAD(_targets);
 static DECLARE_RWSEM(_lock);
 
-#define DM_MOD_NAME_SIZE 32
-
 static inline struct target_type *__find_target_type(const char *name)
 {
 	struct target_type *tt;
-- 
2.16.3

