From: Matthias Brugger <mbrugger@suse.com>
Date: Mon, 6 Feb 2018 17:59:56 +0000
Subject: arm64: add PTE_ADDR_MASK
Patch-mainline: Never, fixes SUSE kernel compilation
References: bsc#1068032

Signed-off-by: Matthias Brugger <mbrugger@suse.com> 
---
 arch/arm64/include/asm/pgtable-hwdef.h |    3 +++
 1 file changed, 3 insertions(+)

--- a/arch/arm64/include/asm/pgtable-hwdef.h
+++ b/arch/arm64/include/asm/pgtable-hwdef.h
@@ -166,6 +166,9 @@
 #define PTE_UXN			(_AT(pteval_t, 1) << 54)	/* User XN */
 #define PTE_HYP_XN		(_AT(pteval_t, 1) << 54)	/* HYP XN */
 
+#define PTE_ADDR_LOW		(((_AT(pteval_t, 1) << (48 - PAGE_SHIFT)) - 1) << PAGE_SHIFT)
+#define PTE_ADDR_MASK          PTE_ADDR_LOW
+
 /*
  * AttrIndx[2:0] encoding (mapping attributes defined in the MAIR* registers).
  */
