From: Christoph Hellwig <hch@lst.de>
Date: Wed, 9 May 2018 09:54:04 +0200
Subject: [PATCH] block: fix __get_request documentation
Git-commit: a9a14d36714fe3669ad5a26faaca0cfaafb1c0f9
Patch-mainline: v4.18-rc1
References: bsc#1104967,FATE#325924

Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Acked-by: Hannes Reinecke <hare@suse.com>
---
 block/blk-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/block/blk-core.c b/block/blk-core.c
index 1418a1ccd80d..e78560f04d77 100644
--- a/block/blk-core.c
+++ b/block/blk-core.c
@@ -1516,7 +1516,7 @@ static struct request *__get_request(struct request_list *rl, unsigned int op,
  * @bio: bio to allocate request for (can be %NULL)
  * @flags: BLK_MQ_REQ_* flags.
  *
- * Get a free request from @q.  If %__GFP_DIRECT_RECLAIM is set in @gfp_mask,
+ * Get a free request from @q.  If %BLK_MQ_REQ_NOWAIT is set in @flags,
  * this function keeps retrying under memory pressure and fails iff @q is dead.
  *
  * Must be called with @q->queue_lock held and,
-- 
2.12.3

