From 043e55d2b7a56163c67a2542597b2e10af8a7ccc Mon Sep 17 00:00:00 2001
From: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
Date: Thu, 3 Jan 2019 12:18:55 +0100
Subject: [PATCH] cpufreq: CPPC: fix build in absence of v3 support

Patch-mainline: Never, Build fix
References: bsc#1117115

Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 drivers/cpufreq/cppc_cpufreq.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/cpufreq/cppc_cpufreq.c b/drivers/cpufreq/cppc_cpufreq.c
index 59ea170..51b61b6 100644
--- a/drivers/cpufreq/cppc_cpufreq.c
+++ b/drivers/cpufreq/cppc_cpufreq.c
@@ -248,6 +248,13 @@ static inline u64 get_delta(u64 t1, u64 t0)
 	return (u32)t1 - (u32)t0;
 }
 
+/* Build fix since CPPCv3 supoort is not included */
+static unsigned int cppc_cpufreq_perf_to_khz(struct cppc_cpudata *cpu,
+                                       unsigned int perf)
+{
+	return (u64)perf * cppc_dmi_max_khz / cpu->perf_caps.highest_perf;
+}
+
 static int cppc_get_rate_from_fbctrs(struct cppc_cpudata *cpu,
 				     struct cppc_perf_fb_ctrs fb_ctrs_t0,
 				     struct cppc_perf_fb_ctrs fb_ctrs_t1)
-- 
2.11.0

