From: Jiri Slaby <jslaby@suse.cz>
Date: Tue, 19 May 2015 16:02:07 +0200
Subject: livepatch: kgr, kgr_in_progress for all threads
Patch-mainline: not yet, kgraft
References: fate#313296 bnc#929883

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 fs/proc/base.c |   23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

--- a/fs/proc/base.c
+++ b/fs/proc/base.c
@@ -2248,6 +2248,26 @@ static int proc_pid_kgr_in_progress(stru
 		struct pid_namespace *ns, struct pid *pid,
 		struct task_struct *task)
 {
+	struct task_struct *t;
+	bool in_progress = false;
+
+	rcu_read_lock();
+	for_each_thread(task, t)
+		if (klp_kgraft_task_in_progress(t)) {
+			in_progress = true;
+			break;
+		}
+	rcu_read_unlock();
+
+	seq_printf(m, "%d\n", in_progress);
+
+	return 0;
+}
+
+static int proc_task_kgr_in_progress(struct seq_file *m,
+		struct pid_namespace *ns, struct pid *pid,
+		struct task_struct *task)
+{
 	seq_printf(m, "%d\n", klp_kgraft_task_in_progress(task));
 
 	return 0;
@@ -3176,6 +3196,9 @@ static const struct pid_entry tid_base_s
 	REG("projid_map", S_IRUGO|S_IWUSR, proc_projid_map_operations),
 	REG("setgroups",  S_IRUGO|S_IWUSR, proc_setgroups_operations),
 #endif
+#if IS_ENABLED(CONFIG_LIVEPATCH)
+	ONE("kgr_in_progress",	S_IRUSR, proc_task_kgr_in_progress),
+#endif
 };
 
 static int proc_tid_base_readdir(struct file *file, struct dir_context *ctx)
