From: Jiri Slaby <jslaby@suse.cz>
Date: Mon, 18 Apr 2016 11:10:59 +0200
Subject: kgr: taint with TAINT_LIVEPATCH
Patch-mainline: not yet, kgraft
References: fate#313296 bsc#974406

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
Acked-by: Jiri Kosina <jkosina@suse.cz>
---
 kernel/kgraft.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/kernel/kgraft.c
+++ b/kernel/kgraft.c
@@ -1127,6 +1127,9 @@ int kgr_patch_kernel(struct kgr_patch *p
 		return -EBUSY;
 	}
 
+	pr_notice_once("tainting kernel with TAINT_LIVEPATCH\n");
+	add_taint(TAINT_LIVEPATCH, LOCKDEP_STILL_OK);
+
 	init_completion(&patch->finish);
 
 	ret = kgr_patch_dir_add(patch);
