From: Paolo Bonzini <pbonzini@redhat.com>
Date: Fri, 14 Jul 2017 13:38:28 +0200
Subject: KVM: VMX: remove unused field
Patch-mainline: v4.13-rc3
Git-commit: fa19871a166f6a940eb97dd511d3ddb1841ed95a
References: bsc#1077761

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Acked-by: Alexander Graf <agraf@suse.de>
---
 arch/x86/kvm/vmx.c |    1 -
 1 file changed, 1 deletion(-)

--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -561,7 +561,6 @@
 	struct kvm_vcpu       vcpu;
 	unsigned long         host_rsp;
 	u8                    fail;
-	bool                  nmi_known_unmasked;
 	u32                   exit_intr_info;
 	u32                   idt_vectoring_info;
 	ulong                 rflags;
