From: Joseph Salisbury <Joseph.Salisbury@microsoft.com>
Date: Tue, 23 Apr 2019 03:37:07 +0000
Patch-mainline: v5.2-rc1
References: fate#321895
Subject: Input: hyperv-keyboard - add module description
Git-commit: 9d1c2f067e50bebb521ed6ced49ddf53bb1deba8

Signed-off-by: Joseph Salisbury <joseph.salisbury@microsoft.com>
Signed-off-by: Dmitry Torokhov <dmitry.torokhov@gmail.com>
Acked-by: Olaf Hering <ohering@suse.de>
---
 drivers/input/serio/hyperv-keyboard.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/input/serio/hyperv-keyboard.c b/drivers/input/serio/hyperv-keyboard.c
--- a/drivers/input/serio/hyperv-keyboard.c
+++ b/drivers/input/serio/hyperv-keyboard.c
@@ -440,5 +440,7 @@ static void __exit hv_kbd_exit(void)
 }
 
 MODULE_LICENSE("GPL");
+MODULE_DESCRIPTION("Microsoft Hyper-V Synthetic Keyboard Driver");
+
 module_init(hv_kbd_init);
 module_exit(hv_kbd_exit);
