From: Keith Busch <keith.busch@intel.com>
Date: Thu, 12 Apr 2018 09:16:14 -0600
Subject: [PATCH] nvme: Use admin command effects for admin commands
References: bsc#1098706
Git-commit: 62843c2e4226057c83f520c74fe9c81a1891c331
Patch-mainline: v4.17-rc1

Signed-off-by: Keith Busch <keith.busch@intel.com>
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/nvme/host/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/nvme/host/core.c b/drivers/nvme/host/core.c
index 88286ad1e0fe..54c69f941dfe 100644
--- a/drivers/nvme/host/core.c
+++ b/drivers/nvme/host/core.c
@@ -1104,7 +1104,7 @@ static u32 nvme_passthru_start(struct nvme_ctrl *ctrl, struct nvme_ns *ns,
 	}
 
 	if (ctrl->effects)
-		effects = le32_to_cpu(ctrl->effects->iocs[opcode]);
+		effects = le32_to_cpu(ctrl->effects->acs[opcode]);
 	else
 		effects = nvme_known_admin_effects(opcode);
 
-- 
2.12.3

