From: Nitzan Carmi <nitzanc@mellanox.com>
Date: Sun, 22 Oct 2017 09:37:04 +0000
Subject: nvme-rdma: Add debug message when reaches timeout
Patch-mainline: v4.15-rc1
Git-commit: e62a538da2e7401bad5293d210ce00583ac12f39
References: FATE#323952, FATE#322506

Signed-off-by: Nitzan Carmi <nitzanc@mellanox.com>
Reviewed-by: Max Gurtovoy <maxg@mellanox.com>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/nvme/host/rdma.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/nvme/host/rdma.c
+++ b/drivers/nvme/host/rdma.c
@@ -1586,6 +1586,10 @@ nvme_rdma_timeout(struct request *rq, bo
 {
 	struct nvme_rdma_request *req = blk_mq_rq_to_pdu(rq);
 
+	dev_warn(req->queue->ctrl->ctrl.device,
+		 "I/O %d QID %d timeout, reset controller\n",
+		 rq->tag, nvme_rdma_queue_idx(req->queue));
+
 	/* queue error recovery */
 	nvme_rdma_error_recovery(req->queue->ctrl);
 
