From: Bart Van Assche <bart.vanassche@wdc.com>
Date: Thu, 5 Jul 2018 10:52:25 -0700
Subject: RDMA/ocrdma: Remove a set-but-not-used variable
Patch-mainline: v4.19-rc1
Git-commit: aa9d5ffbb72746dd1de9d92be7e8b437c41cf0a3
References: bsc#1103992 FATE#326009

Signed-off-by: Bart Van Assche <bart.vanassche@wdc.com>
Signed-off-by: Jason Gunthorpe <jgg@mellanox.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/hw/ocrdma/ocrdma_ah.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/infiniband/hw/ocrdma/ocrdma_ah.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_ah.c
@@ -167,13 +167,11 @@ struct ib_ah *ocrdma_create_ah(struct ib
 	const struct ib_gid_attr *sgid_attr;
 	struct ocrdma_pd *pd = get_ocrdma_pd(ibpd);
 	struct ocrdma_dev *dev = get_ocrdma_dev(ibpd->device);
-	const struct ib_global_route *grh;
 
 	if ((attr->type != RDMA_AH_ATTR_TYPE_ROCE) ||
 	    !(rdma_ah_get_ah_flags(attr) & IB_AH_GRH))
 		return ERR_PTR(-EINVAL);
 
-	grh = rdma_ah_read_grh(attr);
 	if (atomic_cmpxchg(&dev->update_sl, 1, 0))
 		ocrdma_init_service_level(dev);
 
