From: Bart Van Assche <bart.vanassche@wdc.com>
Date: Mon, 12 Feb 2018 08:58:18 -0800
Subject: scsi: pmcraid: Remove an unused structure member
Patch-mainline: v4.17-rc1
Git-commit: f5594686f5f06637bea144159dfe147dae3040d2
References: FATE#325926

Signed-off-by: Bart Van Assche <bart.vanassche@wdc.com>
Reviewed-by: Johannes Thumshirn <jthumshirn@suse.de>
Reviewed-by: Hannes Reinecke <hare@suse.com>
Cc: Anil Ravindranath <anil_ravindranath@pmc-sierra.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
---
 drivers/scsi/pmcraid.h |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/scsi/pmcraid.h
+++ b/drivers/scsi/pmcraid.h
@@ -542,7 +542,6 @@ struct pmcraid_sglist {
 	u32 order;
 	u32 num_sg;
 	u32 num_dma_sg;
-	u32 buffer_len;
 	struct scatterlist scatterlist[1];
 };
 
