From: Anders Roxell <anders.roxell@linaro.org>
Date: Tue, 10 Apr 2018 14:24:21 +0200
Subject: selftests: bpf: update .gitignore with missing generated files
Patch-mainline: v4.17-rc3
Git-commit: 0abf854d7cbbb405e39e0f93d5c1da98dca24bc0
References: bsc#1109837

Signed-off-by: Anders Roxell <anders.roxell@linaro.org>
Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 tools/testing/selftests/bpf/.gitignore |    3 +++
 1 file changed, 3 insertions(+)

--- a/tools/testing/selftests/bpf/.gitignore
+++ b/tools/testing/selftests/bpf/.gitignore
@@ -12,3 +12,6 @@ test_tcpbpf_user
 test_verifier_log
 feature
 test_libbpf_open
+test_sock
+test_sock_addr
+urandom_read
