From: Jan Kara <jack@suse.cz>
Subject: bdi: Preserve kabi when adding cgwb_release_mutex
Patch-mainline: Never, kABI
References: bsc#1112746

Signed-off-by: Jan Kara <jack@suse.cz>

---
 include/linux/backing-dev-defs.h |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/include/linux/backing-dev-defs.h
+++ b/include/linux/backing-dev-defs.h
@@ -187,7 +187,6 @@ struct backing_dev_info {
 #ifdef CONFIG_CGROUP_WRITEBACK
 	struct radix_tree_root cgwb_tree; /* radix tree of active cgroup wbs */
 	struct rb_root cgwb_congested_tree; /* their congested states */
-	struct mutex cgwb_release_mutex;  /* protect shutdown of wb structs */
 #else
 	struct bdi_writeback_congested *wb_congested;
 #endif
@@ -202,6 +201,11 @@ struct backing_dev_info {
 	struct dentry *debug_dir;
 	struct dentry *debug_stats;
 #endif
+#ifdef CONFIG_CGROUP_WRITEBACK
+#ifndef __GENKSYMS__
+	struct mutex cgwb_release_mutex;  /* protect shutdown of wb structs */
+#endif
+#endif
 };
 
 enum {
