From: Gary Lin <glin@suse.com>
Subject: [PATCH] kabi: mask raw in struct bpf_reg_state
Patch-mainline: No, SUSE-specific
References: bsc#1083647

Signed-off-by: Gary Lin <glin@suse.com> 

---
 include/linux/bpf_verifier.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/linux/bpf_verifier.h
+++ b/include/linux/bpf_verifier.h
@@ -51,8 +51,10 @@ struct bpf_reg_state {
 		 */
 		struct bpf_map *map_ptr;
 
+#ifndef __GENKSYMS__
 		/* Max size from any of the above. */
 		unsigned long raw;
+#endif
 	};
 	/* Fixed part of pointer offset, pointer types only */
 	s32 off;
