From: Gary Lin <glin@suse.com>
Subject: [PATCH] kabi: ignore struct bpf_insn_aux_data change
Patch-mainline: No, SUSE-specific
References: bsc#1068032 CVE-2017-5753

Signed-off-by: Gary Lin <glin@suse.com>
---
 include/linux/bpf_verifier.h |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/include/linux/bpf_verifier.h
+++ b/include/linux/bpf_verifier.h
@@ -106,7 +106,9 @@ struct bpf_verifier_state {
 	struct bpf_verifier_state *parent;
 	int allocated_stack;
 	struct bpf_stack_state *stack;
+#ifndef __GENKSYMS__
 	bool speculative;
+#endif
 };
 
 /* linked list of verifier states used to prune search */
@@ -127,12 +129,16 @@ struct bpf_insn_aux_data {
 	union {
 		enum bpf_reg_type ptr_type;	/* pointer type for load/store insns */
 		unsigned long map_state;	/* pointer/poison value for maps */
+#ifndef __GENKSYMS__
 		u32 alu_limit;			/* limit for add/sub register with pointer */
+#endif
 	};
 	int ctx_field_size; /* the ctx field size for load insn, maybe 0 */
 	int sanitize_stack_off; /* stack slot to be cleared */
 	bool seen; /* this insn was processed by the verifier */
+#ifndef __GENKSYMS__
 	u8 alu_state; /* used in combination with alu_limit */
+#endif
 };
 
 #define MAX_USED_MAPS 64 /* max number of maps accessed by one eBPF program */
