From: Jiri Kosina <jkosina@suse.cz>
Subject: [PATCH] kabi: ignore struct bpf_insn_aux_data change
Patch-mainline: No, SUSE-specific
References: bsc#1087082

Signed-off-by: Jiri Kosina <jkosina@suse.cz>
---
 include/linux/bpf_verifier.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/include/linux/bpf_verifier.h
+++ b/include/linux/bpf_verifier.h
@@ -127,9 +127,11 @@ struct bpf_insn_aux_data {
 		u32 alu_limit;			/* limit for add/sub register with pointer */
 	};
 	int ctx_field_size; /* the ctx field size for load insn, maybe 0 */
-	int sanitize_stack_off; /* stack slot to be cleared */
 	bool seen; /* this insn was processed by the verifier */
 	u8 alu_state; /* used in combination with alu_limit */
+#ifndef __GENKSYMS__
+	int sanitize_stack_off; /* stack slot to be cleared */
+#endif
 };
 
 #define MAX_USED_MAPS 64 /* max number of maps accessed by one eBPF program */
