From: Gary Lin <glin@suse.com>
Subject: [PATCH] kabi: ignore struct bpf_insn_aux_data change
Patch-mainline: No, SUSE-specific
References: bsc#1068032 CVE-2017-5753

Signed-off-by: Gary Lin <glin@suse.com>
---
 include/linux/bpf_verifier.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/include/linux/bpf_verifier.h
+++ b/include/linux/bpf_verifier.h
@@ -106,7 +106,9 @@ struct bpf_verifier_state {
 	struct bpf_verifier_state *parent;
 	int allocated_stack;
 	struct bpf_stack_state *stack;
+#ifndef __GENKSYMS__
 	bool speculative;
+#endif
 };
 
 /* linked list of verifier states used to prune search */
@@ -134,9 +136,9 @@ struct bpf_insn_aux_data {
 	};
 	int ctx_field_size; /* the ctx field size for load insn, maybe 0 */
 	bool seen; /* this insn was processed by the verifier */
-	u8 alu_state; /* used in combination with alu_limit */
 #ifndef __GENKSYMS__
 	int sanitize_stack_off; /* stack slot to be cleared */
+	u8 alu_state; /* used in combination with alu_limit */
 #endif
 };
 
