From: Gary Lin <glin@suse.com>
Subject: [PATCH] kabi: add struct bpf_map back
Patch-mainline: No, SUSE-specific
References: References: bsc#1098425

Signed-off-by: Gary Lin <glin@suse.com> 

---
 include/linux/bpf_verifier.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/include/linux/bpf_verifier.h
+++ b/include/linux/bpf_verifier.h
@@ -112,7 +112,11 @@ struct bpf_verifier_state_list {
 struct bpf_insn_aux_data {
 	union {
 		enum bpf_reg_type ptr_type;	/* pointer type for load/store insns */
+#ifndef __GENKSYMS__
 		unsigned long map_state;	/* pointer/poison value for maps */
+#else
+		struct bpf_map *map_ptr;        /* pointer for call insn into lookup_elem */
+#endif
 	};
 	int ctx_field_size; /* the ctx field size for load insn, maybe 0 */
 	bool seen; /* this insn was processed by the verifier */
