From: Torsten Duwe <duwe@suse.de>
Subject: Cover up kABI breakage due to DH key verification
Patch-mainline: Never, kABI fix
References: bsc#1155331

https://jira.suse.com/browse/ECO-936

Cc: Petr Tesarik <ptesarik@suse.cz>
Cc: Takashi Iwai <tiwai@suse.de>

--- a/include/crypto/dh.h
+++ b/include/crypto/dh.h
@@ -39,11 +39,15 @@
 struct dh {
 	void *key;
 	void *p;
+#ifndef __GENKSYMS__
 	void *q;
+#endif
 	void *g;
 	unsigned int key_size;
 	unsigned int p_size;
+#ifndef __GENKSYMS__
 	unsigned int q_size;
+#endif
 	unsigned int g_size;
 };
 
