From e6a2e77e022f47a68fa81fe271d8054de30f0f74 Mon Sep 17 00:00:00 2001
From: Michal Suchanek <msuchanek@suse.de>
Date: Wed, 3 Apr 2019 14:45:00 +0200
Subject: [PATCH] Fix struct page kABI after adding atomic for ppc

References: bsc#1131326, bsc#1108937
Patch-mainline: no, kabi

Signed-off-by: Michal Suchanek <msuchanek@suse.de>
---
 include/linux/mm_types.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/mm_types.h b/include/linux/mm_types.h
index 2b9589f02601..772fcb7e176f 100644
--- a/include/linux/mm_types.h
+++ b/include/linux/mm_types.h
@@ -57,7 +57,9 @@ struct page {
 
 	/* Second double word */
 	union {
+#ifndef __GENKSYMS__
 		atomic_t pt_frag_refcount; /* powerpc */
+#endif
 		pgoff_t index;		/* Our offset within mapping. */
 		void *freelist;		/* sl[aou]b first free object */
 		/* page_deferred_list().prev	-- second tail page */
-- 
2.20.1

