From: Johannes Thumshirn <jthumshirn@suse.de>
Date: Tue 22 May 11:58:05 CEST 2018
Subject: kabi protect struct acpi_nfit_desc
References: bsc#1091424
Patch-mainline: Never, KABI fixup

Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/acpi/nfit/nfit.h |   15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

--- a/drivers/acpi/nfit/nfit.h
+++ b/drivers/acpi/nfit/nfit.h
@@ -197,21 +197,26 @@ struct acpi_nfit_desc {
 	struct device *dev;
 	u8 ars_start_flags;
 	struct nd_cmd_ars_status *ars_status;
-	struct delayed_work dwork;
+	size_t ars_status_size;
+	struct work_struct work;
 	struct list_head list;
 	struct kernfs_node *scrub_count_state;
-	unsigned int max_ars;
 	unsigned int scrub_count;
 	unsigned int scrub_mode;
-	unsigned int scrub_busy:1;
 	unsigned int cancel:1;
+	unsigned int init_complete:1;
 	unsigned long dimm_cmd_force_en;
 	unsigned long bus_cmd_force_en;
 	unsigned long bus_nfit_cmd_force_en;
-	unsigned int platform_cap;
-	unsigned int scrub_tmo;
 	int (*blk_do_io)(struct nd_blk_region *ndbr, resource_size_t dpa,
 			void *iobuf, u64 len, int rw);
+#ifndef __GENKSYMS__
+	unsigned int platform_cap;
+	unsigned int max_ars;
+	unsigned int scrub_tmo;
+	struct delayed_work dwork;
+	unsigned int scrub_busy:1;
+#endif
 };
 
 enum scrub_mode {
