From: Johannes Thumshirn <jthumshirn@suse.de>
Subject: kabi protect struct kvm_sync_regs
Date: Wed Sep  5 09:14:21 CEST 2018
Patch-mainline: Never, KABI fixup
References: bsc#1106948

Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 arch/s390/include/uapi/asm/kvm.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/s390/include/uapi/asm/kvm.h
+++ b/arch/s390/include/uapi/asm/kvm.h
@@ -262,8 +262,10 @@ struct kvm_sync_regs {
 		struct {
 			__u64 reserved1[2];
 			__u64 gscb[4];
+#ifndef __GENKSYMS__
 			__u64 etoken;
 			__u64 etoken_extension;
+#endif
 		};
 	};
 };
