From: Johannes Thumshirn <jthumshirn@suse.de>
Subject: kabi protect struct ccw_device_private
Date: Thu Aug  2 16:37:35 CEST 2018
References: bsc#1103421
Patch-mainline: Never, KABI fixup

Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/s390/cio/io_sch.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/s390/cio/io_sch.h
+++ b/drivers/s390/cio/io_sch.h
@@ -154,7 +154,6 @@ struct ccw_device_private {
 	unsigned long intparm;	/* user interruption parameter */
 	struct qdio_irq *qdio_data;
 	struct irb irb;		/* device status */
-	int async_kill_io_rc;
 	struct senseid senseid;	/* SenseID info */
 	struct pgid pgid[8];	/* path group IDs per chpid*/
 	struct ccw1 iccws[2];	/* ccws for SNID/SID/SPGID commands */
@@ -167,6 +166,9 @@ struct ccw_device_private {
 	u64 cmb_start_time;		/* clock value of cmb reset */
 	void *cmb_wait;			/* deferred cmb enable/disable */
 	enum interruption_class int_class;
+#ifndef __GENKSYMS__
+	int async_kill_io_rc;
+#endif
 };
 
 #endif
