From: Joerg Roedel <jroedel@suse.de>
Date: Fri, 29 Nov 2019 11:14:38 +0100
Subject: [PATCH] kABI: Fix for "KVM: x86: Introduce vcpu->arch.xsaves_enabled"
Patch-mainline: Never, kABI Fix only
References: bsc#1158066

Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 arch/x86/include/asm/kvm_host.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/include/asm/kvm_host.h b/arch/x86/include/asm/kvm_host.h
index 500fa60bf9b5..749becdb3256 100644
--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -513,7 +513,6 @@ struct kvm_vcpu_arch {
 	u64 ia32_misc_enable_msr;
 	u64 smbase;
 	bool tpr_access_reporting;
-	bool xsaves_enabled;
 	u64 ia32_xss;
 
 	/*
@@ -720,6 +719,9 @@ struct kvm_vcpu_arch {
 	bool l1tf_flush_l1d;
 
 	u64 arch_capabilities;
+#ifndef __GENKSYMS__
+	bool xsaves_enabled;
+#endif
 };
 
 struct kvm_lpage_info {
-- 
2.16.3

