From: Joerg Roedel <jroedel@suse.de>
Date: Fri, 9 Aug 2019 10:38:35 +0200
Subject: [PATCH] kABI: Fix kABI for 'struct amd_iommu'
Patch-mainline: no, kABI fix only
References: bsc#1145010

Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 drivers/iommu/amd_iommu_types.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/iommu/amd_iommu_types.h b/drivers/iommu/amd_iommu_types.h
index 379fe8e31dbd..9a505104b1cf 100644
--- a/drivers/iommu/amd_iommu_types.h
+++ b/drivers/iommu/amd_iommu_types.h
@@ -603,8 +603,10 @@ struct amd_iommu {
 	u32 flags;
 	volatile u64 __aligned(8) cmd_sem;
 
+#ifndef __GENKSYMS__
 	/* IRQ notifier for IntCapXT interrupt */
 	struct irq_affinity_notify intcapxt_notify;
+#endif
 };
 
 static inline struct amd_iommu *dev_to_amd_iommu(struct device *dev)
-- 
2.16.3

