From: Thomas Bogendoerfer <tbogendoerfer@suse.de>
Subject: kabi protect fwnode_handle
Patch-Mainline: never, kABI
References: bsc#1098633

Signed-off-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 include/linux/fwnode.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/linux/fwnode.h
+++ b/include/linux/fwnode.h
@@ -29,7 +29,9 @@ struct fwnode_operations;
 struct fwnode_handle {
 	enum fwnode_type type;
 	struct fwnode_handle *secondary;
+#ifndef __GENKSYMS__
 	const struct fwnode_operations *ops;
+#endif
 };
 
 /**
