From: Joerg Roedel <jroedel@suse.de>
Date: Tue, 23 Oct 2018 16:03:52 +0200
Subject: [PATCH] kABI: Hide get_msr_feature() in kvm_x86_ops
Patch-mainline: no, SUSE-specific
References: bsc#1106240

Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 arch/x86/include/asm/kvm_host.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -1102,11 +1102,10 @@ struct kvm_x86_ops {
 	int (*mem_enc_reg_region)(struct kvm *kvm, struct kvm_enc_region *argp);
 	int (*mem_enc_unreg_region)(struct kvm *kvm, struct kvm_enc_region *argp);
 
-	int (*get_msr_feature)(struct kvm_msr_entry *entry);
-
 
 #ifndef __GENKSYMS__
 	bool (*has_emulated_msr)(int index);
+	int (*get_msr_feature)(struct kvm_msr_entry *entry);
 	struct kvm *(*vm_alloc)(void);
 	void (*vm_free)(struct kvm *);
 #endif
