From: Joerg Roedel <jroedel@suse.de>
Date: Thu, 15 Aug 2019 15:45:34 +0200
Subject: [PATCH] kABI: Fix kABI for x86 pci-dma code
Patch-mainline: No, kABI fix only
References: bsc#1136039

Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 arch/x86/kernel/pci-dma.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/x86/kernel/pci-dma.c b/arch/x86/kernel/pci-dma.c
index 01e469f8186d..9930729e5318 100644
--- a/arch/x86/kernel/pci-dma.c
+++ b/arch/x86/kernel/pci-dma.c
@@ -6,7 +6,9 @@
 #include <linux/gfp.h>
 #include <linux/pci.h>
 #include <linux/kmemleak.h>
+#ifndef __GENKSYMS__
 #include <linux/iommu.h>
+#endif
 
 #include <asm/proto.h>
 #include <asm/dma.h>
-- 
2.16.3

