From: Joerg Roedel <jroedel@suse.de>
Date: Tue, 23 Apr 2019 17:25:00 +0200
Subject: [PATCH] KVM: Fix kABI for AMD SMAP Errata workaround
Patch-mainline: never, kABI workaround
References: bsc#1133149

Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 arch/x86/include/asm/kvm_host.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/x86/include/asm/kvm_host.h b/arch/x86/include/asm/kvm_host.h
index 1f9311659b4d..d6428246aa6d 100644
--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -1112,9 +1112,8 @@ struct kvm_x86_ops {
 	int (*get_msr_feature)(struct kvm_msr_entry *entry);
 	struct kvm *(*vm_alloc)(void);
 	void (*vm_free)(struct kvm *);
-#endif
-
 	bool (*need_emulation_on_page_fault)(struct kvm_vcpu *vcpu);
+#endif
 };
 
 struct kvm_arch_async_pf {
-- 
2.16.3

