From: Joerg Roedel <jroedel@suse.de>
Date: Tue, 23 Apr 2019 17:25:00 +0200
Subject: [PATCH] KVM: Fix kABI for AMD SMAP Errata workaround
Patch-mainline: never, kABI workaround
References: bsc#1133149

Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 arch/x86/include/asm/kvm_host.h |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -1106,12 +1106,11 @@ struct kvm_x86_ops {
 	int (*mem_enc_reg_region)(struct kvm *kvm, struct kvm_enc_region *argp);
 	int (*mem_enc_unreg_region)(struct kvm *kvm, struct kvm_enc_region *argp);
 
-	bool (*need_emulation_on_page_fault)(struct kvm_vcpu *vcpu);
-
 #ifndef __GENKSYMS__
 	struct kvm *(*vm_alloc)(void);
 	int (*get_msr_feature)(struct kvm_msr_entry *entry);
 	void (*vm_free)(struct kvm *);
+	bool (*need_emulation_on_page_fault)(struct kvm_vcpu *vcpu);
 	bool (*has_emulated_msr)(int index);
 #endif
 };
