From: Joerg Roedel <jroedel@suse.de>
Date: Tue, 23 Apr 2019 17:25:00 +0200
Subject: [PATCH] KVM: Fix kABI for AMD SMAP Errata workaround
Patch-mainline: never, kABI workaround
References: bsc#1133149

Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 arch/x86/include/asm/kvm_host.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -1118,7 +1118,9 @@ struct kvm_x86_ops {
 
 	int (*get_msr_feature)(struct kvm_msr_entry *entry);
 
+#ifndef __GENKSYMS__
 	bool (*need_emulation_on_page_fault)(struct kvm_vcpu *vcpu);
+#endif
 };
 
 struct kvm_arch_async_pf {
