From: Takashi Iwai <tiwai@suse.de>
Subject: Fix kABI breakage of kvm_x86_ops due to L1TF patches
Patch-mainline: Never, kABI fix
References: bsc#1089343 CVE-2018-3646

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 arch/x86/include/asm/kvm_host.h |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -707,8 +707,10 @@ struct kvm_vcpu_arch {
 	/* be preempted when it's in kernel-mode(cpl=0) */
 	bool preempted_in_kernel;
 
+#ifndef __GENKSYMS__
 	/* Flush the L1 Data cache for L1TF mitigation on VMENTER */
 	bool l1tf_flush_l1d;
+#endif
 };
 
 struct kvm_lpage_info {
@@ -889,7 +891,6 @@ struct kvm_vcpu_stat {
 	u64 signal_exits;
 	u64 irq_window_exits;
 	u64 nmi_window_exits;
-	u64 l1d_flush;
 	u64 halt_exits;
 	u64 halt_successful_poll;
 	u64 halt_attempted_poll;
@@ -906,6 +907,9 @@ struct kvm_vcpu_stat {
 	u64 irq_injections;
 	u64 nmi_injections;
 	u64 req_event;
+#ifndef __GENKSYMS__
+	u64 l1d_flush;
+#endif
 };
 
 struct x86_instruction_info;
