From: Hannes Reinecke <hare@suse.de>
Date: Tue, 19 Feb 2019 13:43:50 +0100
Subject: [PATCH] nvme: kABI fix for scan_lock
Patch-Mainline: never, SLE15 specific kABI fix
References: bsc#1123882

Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 drivers/nvme/host/nvme.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/nvme/host/nvme.h b/drivers/nvme/host/nvme.h
index cf12f579ea2e..c30e2146324c 100644
--- a/drivers/nvme/host/nvme.h
+++ b/drivers/nvme/host/nvme.h
@@ -146,7 +146,6 @@ struct nvme_ctrl {
 	enum nvme_ctrl_state state;
 	bool identified;
 	spinlock_t lock;
-	struct mutex scan_lock;
 	const struct nvme_ctrl_ops *ops;
 	struct request_queue *admin_q;
 	struct request_queue *connect_q;
@@ -236,6 +235,7 @@ struct nvme_ctrl {
 	struct timer_list anatt_timer;
 	struct work_struct ana_work;
 #endif
+	struct mutex scan_lock;
 #endif
 };
 
-- 
2.16.4

