From: Johannes Thumshirn <jthumshirn@suse.de>
Date: Mon 28 Jan 09:27:37 CET 2019
Subject: kabi fixup struct nvme_subsystem
Patch-mainline: Never, KABI fixup
References: bsc#1110705


Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/nvme/host/nvme.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/nvme/host/nvme.h
+++ b/drivers/nvme/host/nvme.h
@@ -263,9 +263,11 @@ struct nvme_subsystem {
 	u8			cmic;
 	u16			vendor_id;
 	struct ida		ns_ida;
+#ifndef __GENKSYMS__
 #ifdef CONFIG_NVME_MULTIPATH
 	enum nvme_iopolicy	iopolicy;
 #endif
+#endif
 };
 
 /*
