From: Hannes Reinecke <hare@suse.de>
Date: Tue, 19 Feb 2019 13:43:50 +0100
Subject: [PATCH] nvme: kABI fix for scan_lock
Patch-Mainline: never, SLE15 specific kABI fix
References: bsc#1123882

Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 drivers/nvme/host/nvme.h |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/drivers/nvme/host/nvme.h
+++ b/drivers/nvme/host/nvme.h
@@ -146,7 +146,6 @@ struct nvme_ctrl {
 	enum nvme_ctrl_state state;
 	bool identified;
 	spinlock_t lock;
-	struct mutex scan_lock;
 	const struct nvme_ctrl_ops *ops;
 	struct request_queue *admin_q;
 	struct request_queue *connect_q;
@@ -235,6 +234,10 @@ struct nvme_ctrl {
 	u16 maxcmd;
 	int nr_reconnects;
 	struct nvmf_ctrl_options *opts;
+
+#ifndef __GENKSYMS__
+	struct mutex scan_lock;
+#endif
 };
 
 #ifdef CONFIG_NVME_MULTIPATH
