From: Denis Kirjanov <denis.kirjanov@suse.com>
Subject: kABI workaround for new struct srb bitfields
References: bsc#1143706 bsc#1082635 bsc#1154526 bsc#1048942 
Patch-mainline: Never, kABI fixup

Signed-off-by: Denis Kirjanov <denis.kirjanov@suse.com>

---

--- a/drivers/scsi/qla2xxx/qla_def.h
+++ b/drivers/scsi/qla2xxx/qla_def.h
@@ -591,20 +591,16 @@ typedef struct srb {
 	 */
 	uint8_t cmd_type;
 	uint8_t pad[3];
+	atomic_t ref_count;
 	wait_queue_head_t nvme_ls_waitq;
 	struct fc_port *fcport;
 	struct scsi_qla_host *vha;
-	unsigned int abort:1;
-	unsigned int aborted:1;
-	unsigned int completed:1;
-
 	uint32_t handle;
 	uint16_t flags;
 	uint16_t type;
 	const char *name;
 	int iocbs;
 	struct qla_qpair *qpair;
-	struct srb *cmd_sp;
 	struct list_head elem;
 	u32 gen1;	/* scratch */
 	u32 gen2;	/* scratch */
@@ -636,7 +632,11 @@ typedef struct srb {
 	struct kref cmd_kref;	/* need to migrate ref_count over to this */
 	void *priv;
 	unsigned int start_timer:1;
+	unsigned int abort:1;
+	unsigned int aborted:1;
+	unsigned int completed:1;
 	void (*put_fn)(struct kref *kref);
+	struct srb *cmd_sp;
 #endif
 } srb_t;
 
