From: Daniel Wagner <dwagner@suse.de>
Date: Thu, 22 Aug 2019 13:36:28 +0200
Subject: scsi: libsas: kABI protect struct sas_task_slow
Patch-mainline: never, kabi
References: bsc#1137322 bsc#1137323 bsc#1138099 bsc#1138100
---
 include/scsi/libsas.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/scsi/libsas.h
+++ b/include/scsi/libsas.h
@@ -625,7 +625,9 @@ struct sas_task_slow {
 	 */
 	struct timer_list     timer;
 	struct completion     completion;
+#ifndef __GENKSYMS__
 	struct sas_task       *task;
+#endif
 };
 
 #define SAS_TASK_STATE_PENDING      1
